packages/http-signature-utils/src/app.ts

```diff
@@ -6,24 +6,46 @@ import json from 'koa-json'
 import bodyParser from 'koa-bodyparser'
 
 import { parseOrProvisionKey } from './utils/key'
-import { createHeaders } from './utils/headers'
+import { createHeaders, Headers } from './utils/headers'
+import { RequestLike } from '.'
 
-const app = new Koa()
+type AppContext<TResponseBody = unknown> = Koa.ParameterizedContext<
+  Koa.DefaultState,
+  Koa.DefaultContext,
+  TResponseBody
+>
+
+const app = new Koa<Koa.DefaultState, AppContext>()
 const router = new Router()
 
 // Load key
 const privateKey = parseOrProvisionKey(process.env.KEY_FILE)
 
-// Router
-router.post('/', async (ctx): Promise<void> => {
-  const { request, keyId } = ctx.request.body
-  if (!keyId || !request.headers || !request.method || !request.url) {
+type GenerateSignatureRequestBody = {
+  request: RequestLike
+  keyId: string
+}
+
+router.post('/', async (ctx: AppContext<Headers>): Promise<void> => {
```

Comment:
With `koa-bodyparser` using `request.body: unknown`, it's not very straightforward to type out the `request.body` as we would like (especially with a route handler _without_ additional middleware, oddly enough. This is because of some weirdness with the `router.post` types & overrides).

I ended up creating a typeguard to cast from `unknown` to `GenerateSignatureRequestBody`.